src/gsy_e/gsy_e_core/sim_results/endpoint_buffer.py

```diff
@@ -325,19 +327,19 @@ def update_stats(self, area: "Area", simulation_status: str,
             "percentage_completed": int(progress_info.percentage_completed)
         }
 
-        self.results_handler.update(
-            self.area_result_dict, self.flattened_area_core_stats_dict,
-            self.current_market_time_slot_str
-        )
+        if calculate_results:
+            self.results_handler.update(
+                self.area_result_dict, self.flattened_area_core_stats_dict,
+                self.current_market_time_slot_str)
 
-        self.bids_offers_trades.clear()
-
-        if (ConstSettings.GeneralSettings.EXPORT_OFFER_BID_TRADE_HR or
-                ConstSettings.GeneralSettings.EXPORT_ENERGY_TRADE_PROFILE_HR):
-            self.offer_bid_trade_hr.update(area)
+            if (ConstSettings.GeneralSettings.EXPORT_OFFER_BID_TRADE_HR or
+                    ConstSettings.GeneralSettings.EXPORT_ENERGY_TRADE_PROFILE_HR):
+                self.offer_bid_trade_hr.update(area)
 
         self.result_area_uuids = set()
         self.update_results_area_uuids(area)
+
+        self.bids_offers_trades.clear()
```

Comment:
I think this can be inside the `update_offer_bid_trade` method. And since this method is called only by this class, then I think that it is better to make it private, even though it is outside the scope of the ticket I think it would be good to fix